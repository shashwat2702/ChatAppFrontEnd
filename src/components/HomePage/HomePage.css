.pageHeader {
    background-color: #ff4d4d;
    height: 4rem;
    text-align: center;
    padding-top: 1rem;
}
div {
    text-align: center;
}