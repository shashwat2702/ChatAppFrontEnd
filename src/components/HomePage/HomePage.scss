.homePageConatiner {
    display: flex;
    min-height: 700px;
    .activeUsers {
        flex: 1;
        border: 4px solid #4d0099;
        background-color: #281a34;
        border-radius: 10px;
        margin: 10px;
        color: white;
        padding: 10px;
        font-size: 20px;
    }
    .actualMessages {
        flex: 3;
        display: flex;
        flex-direction: column;
        justify-content: space-between;

    }
    .receivedMessages {
        border: 4px solid black;
        flex: 6;
        margin: 10px 0;
        padding: 15px;
        border-radius: 10px;
        color: white;
        font-size: 20px;
    }
    .sendMessageContainer {
        display: flex;
        justify-content: space-between;
        flex: 1;
        .messageInput{
            flex:8;
            margin: 5px 10px;
        }
        .sendButton {
            flex:1;
            height: 3rem;
            font-size: 16px;
            width: 8rem;
        }
        @media screen and (max-width: 800px) {
            flex-direction: column;
        }
    }
}