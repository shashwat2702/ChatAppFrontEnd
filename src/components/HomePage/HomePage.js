import React, { Component, Fragment} from 'react';
import { withRouter } from 'react-router-dom';
import  './HomePage.scss';


class HomePage extends Component {
  componentDidMount() {
    const { isLoggedIn, history } = this.props;
    if(!isLoggedIn){
      history.push('/');
    }
  }
  render() {
    return (
    <Fragment> 
      <p> Secure Page</p>
    </Fragment>
    )
  }
};
export default withRouter(HomePage);
