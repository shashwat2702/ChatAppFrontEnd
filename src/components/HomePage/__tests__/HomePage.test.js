import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from '../HomePage';

jest.mock('react-router-dom');

describe('HomePage snapshot', () => {
  it('should render correctly', () => {
    const HomePageSnap = renderer.create(<HomePage />);
    const HomePageJson = HomePageSnap.toJSON();
    expect(HomePageJson).toMatchSnapshot();
  });
});
