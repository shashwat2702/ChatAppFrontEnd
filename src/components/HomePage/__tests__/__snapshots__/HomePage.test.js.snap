// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot should render correctly 1`] = `
Array [
  <nav
    className="pageHeader"
  >
    <div
      className="navLinks"
    >
      <a
        className="active"
        href="#home"
      >
        Home
      </a>
      <a
        href="#news"
      >
        News
      </a>
      <a
        href="#contact"
      >
        Contact
      </a>
      <a
        href="#about"
      >
        About
      </a>
      <img
        alt="hamburger menu"
        src="iconfinder_menu-alt_134216.svg"
      />
    </div>
  </nav>,
  <div>
    <div
      className="loginForm"
    >
      <ul
        class="tab-group"
      >
        <li
          class="tab"
        />
        <li
          class="tab active"
        />
      </ul>
      <div
        class="tab-content"
      >
        <h1>
          Welcome Back
        </h1>
        <form>
          <div
            class="field-wrap"
          >
            <div
              className="inputComponent"
            >
              <label
                htmlFor="LoginEmail"
              >
                Email ID
                <span
                  className="req"
                >
                  *
                </span>
              </label>
              <input
                autoComplete="off"
                id="LoginEmail"
                placeholder="example@mail.com"
                type="email"
              />
            </div>
          </div>
          <div
            class="field-wrap"
          >
            <div
              className="inputComponent"
            >
              <label
                htmlFor="LoginPassword"
              >
                Password
                <span
                  className="req"
                >
                  *
                </span>
              </label>
              <input
                autoComplete="off"
                id="LoginPassword"
                placeholder="Password"
                type="password"
              />
            </div>
          </div>
          <div
            className="buttonComponent"
          >
            <button
              className="button button-block"
              type="submit"
            >
              Log In
            </button>
          </div>
        </form>
        <div
          id="login"
        />
      </div>
    </div>
  </div>,
  <footer
    className="pageFooter"
  >
    This is the footer section
  </footer>,
]
`;
